lib/jobs/redfish-job.js

```diff
@@ -0,0 +1,226 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    redfish = require('redfish-node');
+
+module.exports = redfishJobFactory;
+di.annotate(redfishJobFactory, new di.Provide('Job.Redfish'));
+di.annotate(redfishJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Util',
+    'Assert',
+    'Promise',
+    '_',
+    'Services.Waterline'
+));
+
+function redfishJobFactory(
+    BaseJob,
+    Logger,
+    util,
+    assert,
+    Promise,
+    _,
+    waterline
+) {
+    var logger = Logger.initialize(redfishJobFactory);
+
+    /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function RedfishJob(options, context, taskId) {
+        RedfishJob.super_.call(this, logger, options, context, taskId);
+        
+        this.routingKey = this.context.graphId;
+        assert.uuid(this.routingKey) ;
+
+        this.concurrent = {};
+        this.maxConcurrent = 2;
+    }
+    util.inherits(RedfishJob, BaseJob);
+    
+    /**
+     * @function initClient
+     * @param settings takes OBM settings from node and initializes a new
+     *                 Redfish API client
+     * @description initiate redfish chassis discovery
+     * @return apiClient object
+     */
+    RedfishJob.prototype.initClient = function (settings) {
+        assert.object(settings,
+            'Missing required OBM settings');
+            
+        var apiClient = new redfish.ApiClient();
+        apiClient.basePath = settings.uri.replace(/\/+$/, '');
+        if (settings.username && settings.password) {
+            var token = new Buffer(
+                settings.username + ':' + settings.password
+            ).toString('base64');
+            apiClient.defaultHeaders.Authorization = 'Basic ' + token;
+            process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+        }
+        return Promise.promisifyAll(new redfish.RedfishvApi(apiClient));
+    };
+
+    /**
+     * @function _run
+     * @description the jobs internal run method
+     */
+    RedfishJob.prototype._run = function run() {
+        var self = this;
+        return waterline.workitems.update({name: "Pollers.Redfish"}, {failureCount: 0})
+        .then(function() {
+            self._subscribeRedfishCommand(self.routingKey, function(data) {
```

Comment:
@benbp I'm following the same convention as all of the RackHD pollers, and this seems minor, so there was no immediate need to break from that convention in this PR.   If we think there is room to improve the poller job design then we should submit a proposal, user story or another PR to address it.